src/sections/layer_and_mask_information_section/layer.rs

```diff
@@ -429,13 +429,20 @@ impl IntoRgba for PsdLayer {
     ///
     /// So we transform the pixel's index based on the layer's left and top
     /// position within the PSD.
-    fn rgba_idx(&self, idx: usize) -> usize {
-        let left_in_layer = idx % self.width() as usize;
-        let left_in_psd = self.layer_properties.layer_left as usize + left_in_layer;
+    fn rgba_idx(&self, idx: usize) -> Option<usize> {
+        let left_in_layer = idx as u16 % self.width();
+        let left_in_psd = self.layer_properties.layer_left + left_in_layer as i32;
 
-        let top_in_psd = idx / self.width() as usize + self.layer_properties.layer_top as usize;
+        let top_in_layer = idx as u16 / self.width();
+        let top_in_psd = self.layer_properties.layer_top + top_in_layer as i32;
```

Comment:
I casted unsigned integer to signed integer,
previously this casted signed integer to unsigned integer, which can occur overflow.